### ISSUE TITLE
Error message implies I need a const impl, but there is no such thing

### ISSUE THREAD

**Issue by mcclure** [rust-lang/rust#106089]
I am using Rust 1.65.0. I run
`cargo new impl_test && cd impl_test && cargo add glam && vi src/main.rs`
and enter this code:

```
use glam::IVec2;

const AXES: [IVec2; 4] = [ -IVec2::Y, IVec2::Y, -IVec2::X, IVec2::X ];

fn main() {
}
```
What I am trying to do is create an array containing the four cardinal directions\*. Rust doesn't like it:
```
error[E0015]: cannot call non-const operator in constants
   --> src\main.rs:3:28
    |
3   | const AXES: [IVec2; 4] = [ -IVec2::Y, IVec2::Y, -IVec2::X, IVec2::X ];
    |                            ^^^^^^^^^
    |
note: impl defined here, but it is not `const`
   --> C:\Users\Andi\.cargo\registry\src\github.com-1ecc6299db9ec823\glam-0.22.0\src\i32\ivec2.rs:628:1
    |
628 | impl Neg for IVec2 {
    | ^^^^^^^^^^^^^^^^^^
    = note: calls in constants are limited to constant functions, tuple structs and tuple variants
```
Oh!, I think. The impl is not const! I will go file a issue on glam and ask them to make this impl const.

But then after a moment I think to check, and I realize [const impls are an experimental feature](https://github.com/rust-lang/rust/issues/67792). It is not stable and the RFC has not even been accepted yet. So that would have been a very unhelpful issue.

The error message seems to be a generic one (the method is not const) but **the wording implies that you can fix it by making the impl const**. In other words it implies the problem is with your impl, but actually the problem is impls are not adequate for what you want them to do. (And if this is something that works in nightly, someone might have the unpleasant experience of thinking the compiler told them to add a const to their impl, adding a const to their impl, and then getting a "sorry, unstable" error message).

**"Ideally the output should look like:"**

I would probably word it like
```
note: impl defined here, but impl methods are not `const`
```
or to be even more explicit
```
note: impl defined here, but impl methods can never be `const`
```

If const impls work in nightly, a `hint: const impl is allowed in nightly [tracking issue link]` could be helpful to in addition to the wording change (or could be used in place of the above wording completely)

\* *What I should have done in this case is use IVec2::NEG_X and IVec2::NEG_Y. But that's not the point.*

**Comment by mejrs** [comment]
> the wording implies that you can fix it by making the impl const

I don't think this message is suggestive; there's nothing like "consider making this trait impl const".

**Comment by clubby789** [comment]
const impls do work in nightly, as they're used a lot by the standard library. However, (as I understand it), the existing feature is likely to be superseded and never stabilized, so pushing people toward it with a link wouldn't be useful - especially if the user isn't actually trying to `impl const` and the compiler isn't suggesting they do.

**Comment by Boiethios** [comment]
> const impls do work in nightly, as they're used a lot by the standard library. However, (as I understand it), the existing feature is likely to be superseded and never stabilized, so pushing people toward it with a link wouldn't be useful - especially if the user isn't actually trying to `impl const` and the compiler isn't suggesting they do.

What if I try and do that? The message is confusing (not sure if I should open a new issue):

```rust
#![feature(const_trait_impl)]

pub struct Foo(i32);

impl const std::ops::Add for Foo {
    type Output = Foo;
    
    fn add(self, rhs: Foo) -> Self {
        Foo(self.0 + rhs.0)
    }
}

pub const TEST: Foo = Foo(1) + Foo(2);
```

Message:

```
error[E0015]: cannot call non-const operator in constants
  --> src/lib.rs:13:23
   |
13 | pub const TEST: Foo = Foo(1) + Foo(2);
   |                       ^^^^^^^^^^^^^^^
   |
note: impl defined here, but it is not `const`
  --> src/lib.rs:5:1
   |
5  | impl const std::ops::Add for Foo {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: calls in constants are limited to constant functions, tuple structs and tuple variants
```

The `it` in the message refers to the operator (I think) but it looks like it refers to the implementation.
